### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by shadabwahidullah**
## In this milestone I have:

- Created a class for MyList
- Created a module for MyEnumerable

## Used Technologies:

- Ruby
- Linters

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'my_enumerable'
+
+class MyList
+  include MyEnumerable
+  def initialize(*args)
+    @list = args
+  end
+end
```
Comment (helman101) [inline my_list.rb:8]: - [x] Please remember that you are supposed to implement your own `each` method, you should not use the `each` method given by ruby, this projet is supposed to help you understand how enumerables work, kinldy change this and implement the new `each`method created by you, on the other enumerables 😄
    - You could create your own `each` method using `while` (for example).

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      return true if yield e
+    end
+    false
+  end
+
+  def filter
+    filtered = []
+    @list.each do |e|
+      filtered.push(e) if yield e
+    end
```
Comment (helman101) [inline my_enumerable.rb]: - [x] Using the `each` method given by ruby is not allowed kindly implement your own `each` method in `MyList` class 😄

File: my_enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    @list.each do |e|
+      return false unless yield e
+    end
+    true
+  end
+
+  def any?
+    @list.each do |e|
+      return true if yield e
+    end
```
Comment (helman101) [inline my_enumerable.rb]: - [x] using the `each` method given by ruby is not allowed kindly implement your own `each` method in `MyList` class

File: my_enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+module MyEnumerable
+  def all?
+    @list.each do |e|
+      return false unless yield e
+    end
```
Comment (helman101) [inline my_enumerable.rb]: - [x] using the `each` method given by ruby is not allowed kindly implement your own `each` method in `MyList` class

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @shadabwahidullah 👋, 

Good job so far with this project. 👏 👏👏
You've done well implementing some of the requirements, but there are still some which aren't addressed yet.

## Required Changes ♻️

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+def each(&block)
+  @list.each(&block)
```
Comment (helman101) [inline my_list.rb]: - [x] You have done good progress, unfortunately, you are sill using the `each` method given by ruby, please create your own `each` method without using it, try using `while` to iterate trough the array of arguments 😄

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @shadabwahidullah 👋, 

Great work on making the changes requested by a previous reviewer 👏 👏👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 5 above)
Comment (shadabwahidullah) [reply my_list.rb]: could you provide my link to an example or the documentation, please?

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Wahidullah, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have no more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**